### DISCUSSION TITLE
Easy source for parts?

### DISCUSSION THREAD

**Discussion by ktiedt** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#249] [Q&A] [answered]
I can easily grab the ESP off Amazon along with a 8 channel LLC but the 6 pin connectors and everything from any of the BOM lists appears to no longer work to find the more finicky parts... Is it just me? EasyAda loves to make me login, but hates to show me anything but the home page when I click any links related to downloading the BOM or assembling an order at the parts warehouse either... 

Would love to build and contribute back to the community but, not being someone who orders electronic components in recent years, this has been a bit of an ordeal, and I recently moved to an area where I also don't have easy access to a local store for components... First time in decades living in a pretty rural area, bit of adjusting to do :/ 

Thanks in advance if anyone can shed some light on this... worst case I'll resume hunting for individual parts again tomorrow.

**Comment by visualapproach** [comment, marked as answer]
Hi @ktiedt 
The BOM lists are automatically added when uploading to easyeda and only contains what's in the drawing, so they are not intended (by me) to be used as a one click option. Even if it would be nice. But you can order the PCB with a few clicks on that page. I just downloaded the gerber files to code/PCB/ folder if you fancy another fab.
Other parts like the connectors would need to be sourced separately.
Good luck with your build!

**Reply by ktiedt** [reply]
Ahh thanks for the clarification @visualapproach with the purchase items button it certainly seemed like that was an intended feature of the project. I had hoped that perhaps a known good source of the items was included, as I know parts quality can vary widely from one vendor to another, but I'll manage. 

Thank you again for an amazing project, we snagged our spa on a whim from Amazon (treasure truck sale) and its turned out to be one of our best purchases ever.... and with this project, it can only get better :)

**Comment by mechanimal82** [comment]
Apologies to jump on this, but I thought this was relevant enough o continue this discussion rather than start anew. I'm new to this and am looking to upgrade my dumb Helsinki square pump to link it to my HA. I've got as far as ordering from jlcpcb but want to be sure I'm doing the right thing to match the instructions. Am I correct in thinking that I need to accept the below order (once the out of stock parts are available)? Once done I just need to add the NodeMCU module and cables then I'm good to go? I do have the LLC in my Amazon cart - but it looks like this is added to the board so I think this comes as part of the complete build from jlc - is this correct?

FYI I'm in the UK, but not sure that makes a difference except for the shipping cost :)

TL;DR - Wait for the out of stock parts to be available, order the below from jlcpcb. Solder the cables as per the diagram and attach the NodeMCU with pins. Connect to pump - Happy days!

Edit - If anyone like me was struggling to find the cable; [here](https://www.wish.com/search/jst%20connector%206%20pin/product/60140c5e14775c58b0808221?source_feed_page_view_id=1650332085_851f72a0af584b8597bc927f1fbc475a_SearchPage&source=search&position=1&share=web)'s a link to the one shown in the instructions. Just make sure to change the size as required (4 or 6 pin).

![image](https://user-images.githubusercontent.com/854263/163900561-053a376d-ecdc-4a7a-931e-4552d581748e.png)

**Reply by visualapproach** [reply]
Hi, don't use the BOM. Just order the PCB and source the rest manually.
And be prepared that it may require additional resistors and capacitors as the 2021 has different hardware.

**Reply by mechanimal82** [reply]
Thanks @visualapproach . So to dumb this down a little to my level. I just need to order the fabricated board by itself (no pins, modules or any of the additions in the screenshot) plus the LLC, cables and NodeMCU and then I'm good to go once they're attached to the board?

Thanks, and apologies for being a little dumb :)

**Reply by visualapproach** [reply]
Well yes, I mean you CAN ofc order all from the pcb factory but pick the components by yourself instead. You can buy strings of male/female header pins and cut them to length yourself.

**Reply by mechanimal82** [reply]
Sorry, let me explain. I'm not someone that builds components. I can take things and plug them in; like with a Pi - or directly connecting a wemos via pin cables - but building circuit boards like this is new to me. 

I'm happy to have a go, but it seems I have a technical gap in my understanding. I really want to make my Lay-Z-Spa smarter, but don't want to spend £300 to change to the WiFi pump. This project seemed perfect so I've read the discussions and instructions, but it's still not clear to me. I really appreciate that you're trying to help - but pretend you're speaking to a child :D

What is the easiest, least effort way to build this. Do I need the parts in the screenshot and is it best for the fabricator to do it and me just attach the LLC, cables & MCU?

Or would someone, ideally in the UK be able to build one for me - happy to drop £50 if someone can do it and post to me?

**Reply by mechanimal82** [reply]
OK, I am being an idiot... looking at them they're mainly pins for the board which they put on for you. I've opted for the service but changed the out of stock items for ones that appear to be the same type, I've used their LLC as well - fingers-crossed!

Hopefully when it comes I just need to add the MCU & cables.

**Reply by visualapproach** [reply]
@mechanimal82 Minimum skill requirement is soldering and using a multimeter. There is no preassembled device to buy. There are many variants of pumps and they differ in both how the wires shall be connected, and how the communication works. Some pumps require additional components like resistors to work and that's still not a guarantee it will work.
Now, you haven't mentioned what type of pump you have. If you have a PRE2021 model with the eggshaped 6-wire pump, the device is easy to put together and should not require extra components if you have mentioned skills. Or know someone that has.
Common mistakes are overheating components during soldering, and bad soldering due to not using flux. In these pics you can see the female header pins used between the PCB and the breakout boards. Those will eliminate the risk of overheating while soldering, and will make it possible to swap a bad board if necessary. Use them. The male pins usually comes with the breakout boards and has to be soldered.
Later models has been known to work but appear to need some more skill or luck. Or someone sending me a unit to experiment with :-)

![IMG_3167](https://user-images.githubusercontent.com/22987250/164053848-9f940c50-f132-41ca-a932-297464afd0d1.jpg)
![IMG_3168](https://user-images.githubusercontent.com/22987250/164053872-96370db9-64bb-4296-8091-1f4faa53f38d.jpg)

Good luck

**Reply by mechanimal82** [reply]
Thanks. It's the newer square model pump, let's see what happens when the components turn up. At least if it doesn't work I can just unplug it and use it manually, bit there's no fun in that 😄

**Reply by Skeletorjus** [reply]
@mechanimal82, FWIW, this was my very first soldering project, and I found it pretty easy to do. Worked on first try. Just use female header pins, and you won't risk breaking anything. The minimum order for PCBs when I ordered was 5, and the other components (header pins, ESP8266, LLC and connectors) are cheap enough to order multiple of, so you always have spare parts if needed.

**Reply by ktiedt** [reply]
Well I was going to put this all together today... however, its been a comedy of errors... 1st, the kit I ordered for the 6 pin connectors came with no wire. Then I realized, AFTER I already had flashed and confirmed the firmware was working as needed... the nodemcu I have is NOT the right one either... And it was literally the top result on Amazon for "nodemcu v3" and yet it is too narrow for the PCB.

I am looking forward to offering some improvements along the way though from the repo setup and web UI sides :) (if I ever get all the correct parts anyways...)

**Reply by ktiedt** [reply]
Pretty sure I got all the right bits ordered for Monday... just annoyed I can't even start on this after all this waiting. :P
